flink-formats/flink-avro-apicurio-registry/src/main/java/org/apache/flink/formats/avro/registry/apicurio/ApicurioRegistryAvroFormatFactory.java

```diff
@@ -0,0 +1,289 @@
+        options.add(REGISTERED_ARTIFACT_ID);
+        options.add(REGISTERED_ARTIFACT_VERSION);
+        options.add(PROPERTIES);
+        options.add(SSL_KEYSTORE_LOCATION);
+        options.add(SSL_KEYSTORE_PASSWORD);
+        options.add(SSL_TRUSTSTORE_LOCATION);
+        options.add(SSL_TRUSTSTORE_PASSWORD);
+        options.add(BASIC_AUTH_CREDENTIALS_USERID);
+        options.add(BASIC_AUTH_CREDENTIALS_PASSWORD);
+        options.add(OIDC_AUTH_URL);
+        options.add(OIDC_AUTH_CLIENT_ID);
+        options.add(OIDC_AUTH_CLIENT_SECRET);
+        options.add(OIDC_AUTH_SCOPE);
+        options.add(OIDC_AUTH_TOKEN_EXPIRATION_REDUCTION);
+
+        return options;
+    }
+
+    @Override
+    public Set<ConfigOption<?>> forwardOptions() {
+        return Stream.of(
+                        URL,
+                        USE_HEADERS,
+                        USE_GLOBALID,
+                        SCHEMA,
+                        REGISTERED_ARTIFACT_NAME,
+                        REGISTERED_ARTIFACT_DESCRIPTION,
+                        REGISTERED_ARTIFACT_ID,
+                        REGISTERED_ARTIFACT_VERSION,
+                        PROPERTIES,
+                        SSL_KEYSTORE_LOCATION,
+                        SSL_KEYSTORE_PASSWORD,
+                        SSL_TRUSTSTORE_LOCATION,
+                        SSL_TRUSTSTORE_PASSWORD,
+                        BASIC_AUTH_CREDENTIALS_USERID,
+                        BASIC_AUTH_CREDENTIALS_PASSWORD,
+                        OIDC_AUTH_URL,
+                        OIDC_AUTH_CLIENT_ID,
+                        OIDC_AUTH_CLIENT_SECRET,
+                        OIDC_AUTH_SCOPE,
+                        OIDC_AUTH_TOKEN_EXPIRATION_REDUCTION)
+                .collect(Collectors.toSet());
+    }
+
+    public static @Nullable Map<String, Object> buildOptionalPropertiesMap(
+            ReadableConfig formatOptions) {
+        final Map<String, Object> properties = new HashMap<>();
+
+        Set<ConfigOption<?>> configOptionsArray = getOptionalOptions();
+
+        for (ConfigOption configOption : configOptionsArray) {
+            formatOptions
+                    .getOptional(configOption)
+                    .ifPresent(v -> properties.put(configOption.key(), v));
+            // when there is no value we would like to set the default as the value.
+            // at java 11 we can use the orElse or isEmpty , but we are java 8
+            String configOptionKey = configOption.key();
+            if (properties.get(configOptionKey) == null && configOption.hasDefaultValue()) {
+                properties.put(configOptionKey, configOption.defaultValue());
+            }
+        }
+        return properties;
+    }
+
+    private static Schema getAvroSchema(String schemaString, RowType rowType) {
+        LogicalType convertedDataType =
+                AvroSchemaConverter.convertToDataType(schemaString).getLogicalType();
+
+        if (convertedDataType.isNullable()) {
```

Comment:
Do you make `convertedDataType` non-nullable because `rowType` is always non-nullable, and the `.equals()` below won't work otherwise? A comment here could be helpful.